### DISCUSSION TITLE
Hitchhiker delegation: new mechanism to distribute block proposing better

### DISCUSSION THREAD

**Discussion by ubiubi18** [idena-network/idena-go#1078] [Ideas]
### **Abstract:**
Introduce an additional delegation mechanism that does allow people to delegate to solo-miners and pools without consent while the split share of delegation rewards and mining node owner fees is distributed automatically between both sides. Limit it to 10 slots each mining node.

### **Motivation:** 
The main aim of hitchhiker delegation is to distribute block production better. Identities that dont have the possibility or the will to run a mining node themself but also dont want to delegate to a centralised pool will have a more decentralised alternative to put their proof of person at work to help securing the protocol more effectively. 
 
### **Specification:**
- For each unique mining node address there are up to 10 slots available for "hitchhiker delegation". 
- The mining node owner does not need to give consent and the delegate does not need to ask for consent. 
- You can delegate to complete strangers. The mining node owner has no possibility to terminate the delegate. 
- A hitchhiker delegates vote in oracles can be counted under certain conditions as usual for delegates, but with the difference that the vote of the mining node owner allways outweights the vote of a hitchhiker delegate, no matter at what time it was published.
- The delegation rewards dont go as usual to delegates stake and pools balance, but to delegates stake and beside that to 90% to hitchhiker delegates balance and to 10% to pools/mining node owners balance as fixed compensation. 
- "hitchhiker delegation" and the allready established mechanism for delegation to centralised pools are meant to coexist.

### **Rationale** 
Idena is the most brilliant idea for true decentralisation i know of, but no matter that still has not overcome the issue of monopolised block production. If we look at the block explorer, there are positive tendencies visible towards broader distribution of block production allready thanks to the recent "quadratic" IIPs.  At the time of this writing the top 5 pools in idena propose 41% of all blocks, which may not be meant as attack but if coordinated could probably still be abused to try to manipulate blocks under certain conditions. Generally it is asumed that no player should have the ability to propose more than 1/3 of all blocks to keep the network secure, ideally even far less than that to guarant effective decentralisation at scale. At the time of this writing we have also 2689 validated identities but only 2045 mining identity (delegates and solo-miners) and  486 mining nodes. So 644 identities dont bother to mine or to delegate, this is partially for ignorance or lack of knowledge, partially also for the lack of ability to run a mining node while they are not willing to delegate to a centralised pool at the same time. There are also known people from countries with unstable internet who delegate to big centralised pools simply because they lack alternatives and there is barely an incentive for people to strike deals and even communicate with strangers for just a few idna profit, so this business is left only for big pools. This is a wasted potential that could be used partially with hitchhiker delegation. With hitchhiker delegation the block proposer share of the top 5 centralised pools should shrink further, while it does NOT at all disincentify pool owners to use the delegation mechanism. On the other hand there could be also some people who broker human work that will see more benefit in this mechanism than in working for a pool owner at worse conditions (for example the lead pool did distribute according to its webside only about 40% of rewards to delegates). 

So i would expect this proposal to help Idena to decentralise Block production much better, taking advantage of a so far wasted potential of identities that dont mine, without downsides to security or cryptonomics. Smaller mining nodes would propose blocks more often without questioning the current cryptonomics and voting rights, without needing to make blockproposing "quadratic" as well.

### **Backward Compatibility:**
this change requires a hard fork

### **Security Considerations:**
This proposal does not bear risks for the Idena protocol. 
If the proposed changes cause an unwanted imbalance in the economics of iDNA, it will be possible to revert the changes and return to the previous delegation mechanism by initiating a fork.

**Comment by ubiubi18** [comment]
I understand this proposal has no priority right now, just want to start some discussion. I would be happy if it adds benefits.

**Comment by NPC69** [comment]
I like this idea. It helps raise the level of decentralisation while still making Idena mining convenient for normies. It also makes running a node more profitable.

**Comment by ubiubi2018** [comment]
At the end of last epoch the top 5 pools did propose about 41% of blocks. This has gone only slightly down at the beginning of current epoch, to about 40,5%

**Comment by midenaio** [comment]
@ubiubi18 thanks for the proposal. It aims to reduce the concentration in block production of the large pools. Hitchhiking delegation should motivate dormant miners to delegate their mining power to random non-pool validators. 

However, the proposed change could backfire and reduce the total number of mining nodes. For example, desktop miners who currently turn their nodes on and off daily may instead use hitchhiking delegation permanently.

**Reply by ubiubi2018** [reply]
I did check the blockexplorer for the last 24 hours and did find only 65 "mining off" transactions. But we have at the same time 507 identities that dont mine. So maybe the risk of loosing on/off -miners, while true, could be more than covered by the benefits of more distributed mining power? Also running a solo-mining-node can become more valuable thanks to the reward-share dedicated to node-owner? Anyway, i agree such an idea has some risks attached that need to be considered. Maybe the reward-share should be different?

**Reply by NPC69** [reply]
There could be a way to avoid such unwanted scenarios. It would be better if the node operator could approve potential hitchhikers or make the fee for hitchhiking a node a bit higher, incentivising the on/off people to run their node 24/7.

**Reply by teavar** [reply]
This mechanism could be made dependent from network size, i.e. the less mining nodes the lower % of the mining rewards you get from hitchhiking
